src/order/fixed_points.lean

```diff
@@ -131,62 +124,62 @@ def next (x : α) : α := lfp (λ z, x ⊔ f z)
-calc prev f a = a ⊓ f (prev f a) :
-    gfp_eq $ show monotone (λz, a ⊓ f z), from assume x y h, inf_le_inf_left _ (hf h)
-  ... = f (prev f a) :
-    inf_of_le_right $ le_trans (hf prev_le) h
+lemma prev_eq (hf : monotone f) {a : α} (h : f a ≤ a) : f (prev f a) = prev f a :=
+calc f (prev f a) = a ⊓ f (prev f a) :
+    (inf_of_le_right $ (hf prev_le).trans h).symm
+  ... = prev f a :
+    gfp_fixed_point $ show monotone (λ z, a ⊓ f z), from λ x y h, inf_le_inf_left _ (hf h)
 
 def prev_fixed (hf : monotone f) (a : α) (h : f a ≤ a) : fixed_points f :=
-⟨prev f a, (prev_eq hf h).symm⟩
+⟨prev f a, prev_eq hf h⟩
 
-theorem next_le {x : α} : x ≤ next f x := le_lfp $ λ z hz, le_trans le_sup_left hz
+lemma le_next {x : α} : x ≤ next f x := le_lfp $ λ z hz, le_sup_left.trans hz
 
-lemma next_eq (hf : monotone f) {a : α} (h : a ≤ f a) : next f a = f (next f a) :=
-calc next f a = a ⊔ f (next f a) :
-    lfp_eq $ show monotone (λz, a ⊔ f z), from assume x y h, sup_le_sup_left (hf h) _
- ... = f (next f a) :
-    sup_of_le_right $ le_trans h (hf next_le)
+lemma next_eq (hf : monotone f) {a : α} (h : a ≤ f a) : f (next f a) = next f a :=
+calc f (next f a) = a ⊔ f (next f a) :
+    (sup_of_le_right $ h.trans (hf le_next)).symm
+ ... = next f a :
+    lfp_fixed_point $ show monotone (λ z, a ⊔ f z), from λ x y h, sup_le_sup_left (hf h) _
 
 def next_fixed (hf : monotone f) (a : α) (h : a ≤ f a) : fixed_points f :=
-⟨next f a, (next_eq hf h).symm⟩
+⟨next f a, next_eq hf h⟩
 
 variable f
 
-theorem sup_le_f_of_fixed_points (x y : fixed_points f) : x.1 ⊔ y.1 ≤ f (x.1 ⊔ y.1) :=
+lemma sup_le_f_of_fixed_points (x y : fixed_points f) : x.1 ⊔ y.1 ≤ f (x.1 ⊔ y.1) :=
 sup_le
   (x.2 ▸ (hf $ show x.1 ≤ f x.1 ⊔ y.1, from x.2.symm ▸ le_sup_left))
   (y.2 ▸ (hf $ show y.1 ≤ x.1 ⊔ f y.1, from y.2.symm ▸ le_sup_right))
 
-theorem f_le_inf_of_fixed_points (x y : fixed_points f) : f (x.1 ⊓ y.1) ≤ x.1 ⊓ y.1 :=
+lemma f_le_inf_of_fixed_points (x y : fixed_points f) : f (x.1 ⊓ y.1) ≤ x.1 ⊓ y.1 :=
 le_inf
   (x.2 ▸ (hf $ show f (x.1) ⊓ y.1 ≤ x.1, from x.2.symm ▸ inf_le_left))
   (y.2 ▸ (hf $ show x.1 ⊓ f (y.1) ≤ y.1, from y.2.symm ▸ inf_le_right))
 
-theorem Sup_le_f_of_fixed_points (A : set α) (HA : A ⊆ fixed_points f) : Sup A ≤ f (Sup A) :=
+lemma Sup_le_f_of_fixed_points (A : set α) (HA : A ⊆ fixed_points f) : Sup A ≤ f (Sup A) :=
 Sup_le $ λ x hxA, (HA hxA) ▸ (hf $ le_Sup hxA)
 
-theorem f_le_Inf_of_fixed_points (A : set α) (HA : A ⊆ fixed_points f) : f (Inf A) ≤ Inf A :=
+lemma f_le_Inf_of_fixed_points (A : set α) (HA : A ⊆ fixed_points f) : f (Inf A) ≤ Inf A :=
 le_Inf $ λ x hxA, (HA hxA) ▸ (hf $ Inf_le hxA)
 
-/-- The fixed points of `f` form a complete lattice.
+/-- Knaster-Tarski Theorem: The fixed points of `f` form a complete lattice.
 This cannot be an instance, since it depends on the monotonicity of `f`. -/
 protected def complete_lattice : complete_lattice (fixed_points f) :=
-{ le           := λx y, x.1 ≤ y.1,
+{ le           := λ x y, x.1 ≤ y.1,
```

Comment:
Does it work if you change this to
```suggestion
{ le           := (≤),
```
as you had it in your old version?

Reply:
Ah of course. Let me golf down the rest of the instance.